fullstack-cert/typescript-projects/flashcard-quiz-app/src/script.ts

```diff
@@ -0,0 +1,244 @@
+const cardDisplay = document.querySelector<HTMLElement>("#current-card");
+const cardButtonsContainer = document.querySelector<HTMLElement>("#cards-list");
+const frontInput = document.querySelector<HTMLTextAreaElement>("#front-text");
+const backInput = document.querySelector<HTMLTextAreaElement>("#back-text");
+
+let currentCardIndex = -1;
+let showingFront = true;
+let currentCards: FlashCard[] = [];
+
+interface FlashCard {
+  frontText: string;
+  backText: string;
+}
+
+class InvalidUserInputError extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = "InvalidUserInputError";
+    Object.setPrototypeOf(this, InvalidUserInputError.prototype);
+  }
+}
+
+const isButtonElement = (element: unknown): element is HTMLButtonElement => {
+  return element instanceof HTMLButtonElement;
+};
+
+function refresh(): void {
+  if (currentCardIndex < 0 || !currentCards[currentCardIndex]) {
+    cardDisplay.textContent =
+      "No card selected. Add a new card to get started!";
+    return;
+  }
+
+  const card = currentCards[currentCardIndex];
+  cardDisplay.querySelector(".card-front").textContent = card.frontText;
+  cardDisplay.querySelector(".card-back").textContent = card.backText;
+  // add correct background to current card
+  Array.from(cardButtonsContainer.children).forEach((child, i) => {
+    if (!isButtonElement(child)) {
+      console.warn(`Element {${child}} is not a button.`);
+      return;
+    }
+    const button = child as HTMLElement;
+    if (i === currentCardIndex) {
+      button.classList.add("selected");
+    } else {
+      button.classList.remove("selected");
+    }
+  });
+}
+
+function deleteCard(): void {
+  if (currentCardIndex === -1) {
+    console.log("No card selected to delete.");
+    return;
+  }
+
+  currentCards.splice(currentCardIndex, 1);
+  cardButtonsContainer.removeChild(
+    cardButtonsContainer.children[currentCardIndex],
+  );
+
+  Array.from(cardButtonsContainer.children).forEach((child, i) => {
+    if (!isButtonElement(child)) {
+      console.warn(`Element {${child}} is not a button.`);
+      return;
+    }
+
+    (child as HTMLButtonElement).onclick = () => {
```

Comment:
```suggestion
    child.onclick = () => {
```

The typeguard above makes this cast unnecessary.